default/python/AI/PolicyAI.py

```diff
@@ -0,0 +1,242 @@
+
+    def _can_adopt(self, name, replace_other=None):
+        policy = fo.getPolicy(name)
+        ret = name in self._empire.availablePolicies and policy.adoptionCost() <= self._ip
+        if replace_other and replace_other in self._adopted:
+            return ret
+        return (
+            ret and self._empire.emptyPolicySlots[policy.category] and self._empire.policyPrereqsAndExclusionsOK(name)
+        )
+
+    def _get_adoptable(self):
+        """List of adoptable policies with the still available IP."""
+        ret = set()
+        for p in self._empire.availablePolicies:
+            if p in self._adopted:
+                continue
+            if self._can_adopt(p):
+                ret.add(p)
+        return ret
+
+    def _adopt(self, name, readopt=False):
+        """Find an emtpy slot and adopt named policy, if possible."""
+        policy = fo.getPolicy(name)
+        cat = policy.category
+        slot = self._find_empty_slot(cat)
+        chk = fo.issueAdoptPolicyOrder(name, cat, slot)
+        if chk:
+            debug("Issued adoption order for %s in slot %d turn %d", name, slot, fo.currentTurn())
+            if not readopt:
+                self._ip -= policy.adoptionCost()
+            self._adoptable = self._get_adoptable()
+        else:
+            error("Failed to adopt %s in slot %d", name, slot)
+
+    def _find_empty_slot(self, cat):
+        """This assumes there is an empty slot, otherwise adopt will fail."""
+        slot = 0
+        # there should be a better way...
+        cat_slots = self._empire.categoriesSlotPolicies
+        if cat in cat_slots:
+            slots = cat_slots[cat]
+            for slot in range(0, 99):
+                if slot not in slots:
+                    break
+        return slot
+
+    def _deadopt(self, name):
+        """Deadopt name, if it is adopted."""
+        if name in self._adopted:
+            debug("Issued deadoption order for %s turn %d", name, fo.currentTurn())
+            fo.issueDeadoptPolicyOrder(name)
+            self._adopted.remove(name)
+            self._adoptable = self._get_adoptable()
+
+    def status(self):
+        # only for interactive debugging
+        print("\nAdoptable Policies:")
+        for p in self._get_adoptable():
+            print("  ", p)
+        print("Empty Slots:")
+        for s in self._empire.emptyPolicySlots:
+            print("  ", s)
+        print("Adopted:")
+        for cat in self._empire.categoriesSlotPolicies:
+            if cat:
+                print(" ", cat.key() + ":")
+                for slot in cat.data():
+                    print("   ", slot.key(), "->", slot.data(), "turn", self._empire.turnPolicyAdopted(slot.data()))
+        print("Influence:", self._ip)
+        print("Infl. Prod.:", self._get_infl_prod())
+        print("Num Planet (pop./outp.): %d/%d" % (self._num_populated, self._num_outposts))
+
+
+def generate_policy_orders():
+    PolicyManager().generate_orders()
+
+
+def status():
```

Comment:
Maybe rename it to `print_status` it will highlight debug purposes of this function.